docs/en/reference/types.rst

```diff
@@ -905,3 +905,50 @@ hook it into the database platform:
 This would allow using a money type in the ORM for example and
 have Doctrine automatically convert it back and forth to the
 database.
+
+It is also possible to register type instances directly, in case you
+need to pass parameters to your instance::
+
+    <?php
+    namespace My\Project\Types;
+
+    use Doctrine\DBAL\Types\Type;
+    use Doctrine\DBAL\Platforms\AbstractPlatform;
+
+    final class EncryptedType extends StringType
+    {
+        public function __construct(
+            private string $cipher,
+            private string $secretName,
+            private SecretStore $secretStore
+        ) {
+        }
+
+        public function convertToPHPValue($value, AbstractPlatform $platform): string
+        {
+            return openssl_decrypt(
```

Comment:
IMO, encryption is quote specific and sophisticated domain to be used to illustrate a concept of types. It's better to use a dummy example with a prefix that doesn't distract from the subject.

Reply:
Ok, I tried to dumb it down to a less realistic but more simple use case.